### PR TITLE
Add netplan glide and pwgen

### PR DESCRIPTION
**Pull request by chalamalasetty**
Add netplan, glide and pwgen packages to Mariner.

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
 Add netplan, glide and pwgen packages to Mariner.

###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Local build

### CODE REVIEW

File: SPECS/glide/glide.spec [hunk 1]
```diff
@@ -0,0 +1,87 @@
+Summary:        Vendor Package Management for Goland
```
Comment (anphel31) [inline SPECS/glide/glide.spec:1]: "golang" instead of "goland"?

**Review by anphel31 — APPROVED** [review]

File: SPECS/glide/glide.spec (hunk 1 above)
Comment (chalamalasetty) [reply SPECS/glide/glide.spec:1]: This is inherited from PhotonOS and shows as Goland. Doesn't look like a typo. 
https://github.com/vmware/photon/blob/master/SPECS/glide/glide.spec

**Review by thcrain-msft — APPROVED** [review]
Approving with regards to licensing details, as requested.

File: SPECS/LICENSES-AND-NOTICES/data/licenses.json [hunk 2]
```diff
@@ -862,6 +864,13 @@
         "ceph"
       ]
     },
+    {
+      "origin": "netplan",
+      "license": "[GPLv3](https://github.com/canonical/netplan/blob/main/COPYING)",
```
Comment (jslobodzian) [inline SPECS/LICENSES-AND-NOTICES/data/licenses.json:869]: just to be sure...the netplan SPEC file originated from netplan and not fedora, or centos or something?

File: SPECS/glide/glide.spec [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+install -vpm 0755 -t %{buildroot}%{_bindir}/ ./glide
+popd
+
+%files
+%defattr(-,root,root)
+%{_bindir}/glide
+
+%changelog
+*   Fri Sep 17 2021 Suresh Babu Chalamalasetty <schalam@microsoft.com> - 0.13.3-6
+-   Initial CBL-Mariner import from Photon
```
Comment (jslobodzian) [inline SPECS/glide/glide.spec:56]: Please add after Photon: (license: Apache2) 
That was the license that we pulled the SPEC file under.

**Review by jslobodzian — APPROVED** [review]

File: SPECS/glide/glide.spec [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+%install
+export GO111MODULE=auto
+export GOPATH=%{OUR_GOPATH}
+pushd ${GOPATH}/src/github.com/Masterminds/glide
+make install %{?_smp_mflags}
+install -vdm 755 %{buildroot}%{_bindir}
+install -vpm 0755 -t %{buildroot}%{_bindir}/ ./glide
+popd
+
+%files
```
Comment (hbeberman) [inline SPECS/glide/glide.spec:50]: Glide needs a %license line

File: SPECS/LICENSES-AND-NOTICES/data/licenses.json (hunk 2 above)
Comment (chalamalasetty) [reply SPECS/LICENSES-AND-NOTICES/data/licenses.json:869]: It is from Netplan sources. 
Here is the reference:  https://github.com/canonical/netplan/blob/main/rpm/netplan.spec

File: SPECS/glide/glide.spec (hunk 1 above)
Comment (hbeberman) [reply SPECS/glide/glide.spec:1]: Goland is clearly a typo and we should fix it. It says "Glide: Vendor Package Management for Golang" right at the top of the README

File: SPECS/netplan/netplan.spec [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+Name:           netplan
+Version:        0.95
+Release:        1%{?dist}
+Summary:        Network configuration tool using YAML
+Group:          System Environment/Base
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+License:        GPLv3
+URL:            http://netplan.io/
```
Comment (hbeberman) [inline SPECS/netplan/netplan.spec:22]: nit: Should be https://netplan.io/

File: SPECS/glide/glide.spec (hunk 3 above)
Comment (chalamalasetty) [reply SPECS/glide/glide.spec]: Added.

File: SPECS/netplan/netplan.spec [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+%files
+%license COPYING
+%doc debian/changelog
+%doc %{_docdir}/%{name}/
+%{_sbindir}/%{name}
+%{_datadir}/%{name}/
+%{_unitdir}/%{name}*.service
+%{_systemdgeneratordir}/%{name}
+%{_mandir}/man5/%{name}.5*
+%dir %{_sysconfdir}/%{name}
```
Comment (hbeberman) [inline SPECS/netplan/netplan.spec:101]: Might want to change this to `%dir %attr(0755, root, root)  %{_sysconfdir}/%{name}` to ensure that /etc/netplan is readable to all users like it is on Ubuntu.

I've run into issues recently where some of our /etc/ folders aren't non-root readable and its caused problems with programs trying to read the configs.

File: SPECS/pwgen/pwgen.spec [hunk 7]
```diff
@@ -0,0 +1,138 @@
+Name:           pwgen
+Version:        2.08
+Release:        8%{?dist}
+Summary:        Automatic password generation
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+License:        GPL+
+URL:            http://sf.net/projects/pwgen
+Source0:        http://download.sf.net/pwgen/%{name}-%{version}.tar.gz
```
Comment (hbeberman) [inline SPECS/pwgen/pwgen.spec:9]: Use https for Source0 and URL

**Review by hbeberman — CHANGES_REQUESTED** [review]

File: SPECS/netplan/netplan.spec (hunk 6 above)
Comment (chalamalasetty) [reply SPECS/netplan/netplan.spec]: Thanks Henry. Updated as per your suggestion,

File: SPECS/netplan/netplan.spec (hunk 5 above)
Comment (chalamalasetty) [reply SPECS/netplan/netplan.spec]: added

File: SPECS/glide/glide.spec (hunk 4 above)
Comment (chalamalasetty) [reply SPECS/glide/glide.spec:50]: added

**Review by hbeberman — APPROVED** [review]

**Pull request merged**